packages/core/src/connectivity/scp-cf/authorization-header.ts

```diff
@@ -61,11 +61,21 @@ function hasAuthHeaders(destination: Destination): boolean {
   return authTypesWithAuthorizationHeader.includes(destination.authentication);
 }
 
+function getCustomAuthHeaders(
+  destination: Destination,
+  customHeaders?: Record<string, any>
+) {
+  if (destination.authentication === 'PrincipalPropagation') {
+    return pickIgnoreCase(customHeaders, 'SAP-Connectivity-Authentication');
+  }
+  return pickIgnoreCase(customHeaders, 'authorization');
+}
+
 export async function getAuthHeaders(
   destination: Destination,
   customHeaders?: Record<string, any>
 ): Promise<Record<string, string>> {
-  const customAuthHeaders = pickIgnoreCase(customHeaders, 'authorization');
+  const customAuthHeaders = getCustomAuthHeaders(destination, customHeaders);
 
   if (Object.keys(customAuthHeaders).length && hasAuthHeaders(destination)) {
```

Comment:
the `hasAuthHeaders` is not triggered in the `SAP-Connectivity-Authentication` case right? I would also like to have a warning is this is done.

Reply:
It does. It just checks for authentication type.

Reply:
And we do not want to add a log statement if the `SAP-Connectivity-Authentication` is set via the custom header and is overwriting the automatic one?

Reply:
I meant, that the warning is logged in case there is a custom header.

Reply:
But I'll add a test for this.